libs/vkd3d/resource.c

```diff
@@ -3619,6 +3619,11 @@ HRESULT d3d12_resource_create_committed(struct d3d12_device *device, const D3D12
             allocate_info.flags = VKD3D_ALLOCATION_FLAG_GLOBAL_BUFFER |
                     VKD3D_ALLOCATION_FLAG_ALLOW_IMAGE_SUBALLOCATION;
 
+            /* If image suballocation is allowed force the memory alignment to respect
+             * the device buffer image granularity to prevent resource aliasing */
+            allocate_info.memory_requirements.alignment = max(allocate_info.memory_requirements.alignment,
```

Comment:
This is confusing. We use explicit_global_buffer_usage below there, which should ensure that suballocation from images and buffer never allocate from the same chunk. Did those tests you mention really end up binding a UBO and image from the same VkDeviceMemory?